src/rules/flake8_simplify/rules/fix_if.rs

```diff
@@ -0,0 +1,109 @@
+use anyhow::{bail, Result};
+use libcst_native::{
+    BooleanOp, BooleanOperation, Codegen, CodegenState, CompoundStatement, Expression, If,
+    LeftParen, ParenthesizableWhitespace, ParenthesizedNode, RightParen, SimpleWhitespace,
+    Statement, Suite,
+};
+use rustpython_ast::Location;
+
+use crate::ast::types::Range;
+use crate::cst::matchers::match_module;
+use crate::fix::Fix;
+use crate::source_code::Locator;
+
+fn parenthesize_and_operand(expr: Expression) -> Expression {
+    match expr {
+        Expression::BooleanOperation(boolean_operation)
+            if matches!(boolean_operation.operator, BooleanOp::Or { .. }) =>
+        {
+            Expression::BooleanOperation(
+                boolean_operation.with_parens(LeftParen::default(), RightParen::default()),
+            )
+        }
+        Expression::IfExp(if_exp) => {
+            Expression::IfExp(if_exp.with_parens(LeftParen::default(), RightParen::default()))
+        }
+        Expression::Lambda(lambda) => {
+            Expression::Lambda(lambda.with_parens(LeftParen::default(), RightParen::default()))
+        }
+        Expression::NamedExpr(named_expr) => Expression::NamedExpr(
+            named_expr.with_parens(LeftParen::default(), RightParen::default()),
+        ),
+        _ => expr,
+    }
+}
+
+/// (SIM102) Convert `if a: if b:` to `if a and b:`.
+pub(crate) fn fix_nested_if_statements(
+    locator: &Locator,
+    stmt: &rustpython_ast::Stmt,
+) -> Result<Fix> {
+    // Expand the start and end location of the fix to include leading and trailing
+    // whitespace.
+    let start = Location::new(stmt.location.row(), 0);
+    let end = Location::new(stmt.end_location.unwrap().row() + 1, 0);
+
+    // Extract and dedent contents.
+    let contents = locator.slice_source_code_range(&Range::new(start, end));
+
+    // Store and validate the statement indentation.
+    let indentation = contents
+        .chars()
+        .take(stmt.location.column())
+        .collect::<String>();
+    if !indentation.chars().all(char::is_whitespace) {
+        bail!("Unable to fix multiline statements");
+    }
+
+    let contents = textwrap::dedent(&contents);
```

Comment:
This has various failure modes.

```python
while True:
    if True:
        if True:
            """this
is valid"""

            """the indentation on
            this line is significant"""

            "this is" \
"allowed too"

            ("so is"
"this for some reason")
```

Reply:
Right, ok, good call. I believe I can fix this by leveraging LibCST's `indentation` field.

Reply:
Alright, I think the new version works, but it requires a bunch of trickery.

If the statement is indented, we embed it in a dummy function, like:

```py
def f():
  if ...
```

Then, at the end, we stripe the function. I _think_ this is the most robust way to preserve indentation. It lets us avoid having to do any indentation or dedentation ourselves (but keeps the source code valid for LibCST).

Separately, we _also_ have to handle `elif` blocks via special-case.